requirements_parser/requirements_parser.py

```diff
@@ -0,0 +1,40 @@
+class RequirementsParser:
+
+    def open_requirements(self, path):
+        """
+        adding to dict all the dependencies
+
+        Parameters:
+        file (str): requirements.txt file location / requirements.txt file name
+
+        Returns:
+        out_dict(dict): contains all the dependencies
+
+        """
+        output_dict = set()
+        with open(path, "r") as requirements:
+            listed_file = [line.strip() for line in requirements if not line == "\n"]
+
+        for line in listed_file:
+            if "-r" in line:
+                line = line.removeprefix('-r ')
+                output_dict.update(self.open_requirements(line))
+                print(output_dict)
+            else:
+                output_dict.add(line)
+        return output_dict
+
+    def create_final_requirements_filename(self, path):
```

Comment:
why the function doesn't get file_name, why path?

Reply:
by the way.. if you desice that is get path.. better you call it requirment_file_path